tflog/provider.go

```diff
@@ -104,5 +130,142 @@ func Error(ctx context.Context, msg string, additionalFields ...map[string]inter
 		// so just making this a no-op is fine
 		return
 	}
-	logger.Error(msg, hclogutils.MapsToArgs(additionalFields...)...)
+
+	additionalArgs, shouldOmit := omitOrMask(ctx, logger, &msg, additionalFields)
+	if shouldOmit {
+		return
+	}
+
+	logger.Error(msg, additionalArgs...)
+}
+
+func omitOrMask(ctx context.Context, logger hclog.Logger, msg *string, additionalFields []map[string]interface{}) ([]interface{}, bool) {
+	tfLoggerOpts := logging.GetProviderRootTFLoggerOpts(ctx)
+	additionalArgs := hclogutils.MapsToArgs(additionalFields...)
+	impliedArgs := logger.ImpliedArgs()
```

Comment:
Doing the filtering on the maps, would force the logic to convert the `ImpliedArgs` to map, before it can be filtered.
So I did the opposite, given that the underlying library requires to go in the `args = [][]interface{}` direction.

Reply:
I agree that the `ImpliedArgs()` (that ultimately dictate this) are a sub-par thing from `hclog`. But I didn't want to expand my work to have to also alter `With()` so to always keep those key/value pairs stored in the context.

And writing the couple of `i += 2` for loops was relatively easy.

We can refactor that later I guess.

Reply:
There was no need to refactor With() _now_, we could've introduced the slice to map converter now, but like you say, it can be refactored later.

Reply:
I'll create a follow up ticket to do the refactoring and simplification: iterating over the keys of a map is definitely the most desirable way to go. Just focused on getting the functionality done and working on this one.